### PR TITLE
feat(issues): Add new IssueCategoryFilter component to Issues Stream

### PR DESCRIPTION
**Pull request by 0Calories**
Adds a new dropdown button on the issues stream to allow for quickly filtering issues by their category, i.e `Performance` and `Error`.

### Features:
- Selecting a dropdown option will immediately filter all issues by the selected category
- Selecting a dropdown option will also add its respective query tag to the smart search bar, i.e `issue.category:performance`
- Manually updating the query in the smart search bar will cause the dropdown to update its active option, i.e searching for `issue.category:error` will cause the dropdown option to be set to `Errors`.
- The `Performance` option in the dropdown has a `new` badge until it is selected for the first time
- Sends analytics events when the dropdown option is changed (only when the dropdown is directly interacted with)

Reload PR For Analytics: https://github.com/getsentry/reload/pull/308

### Demo
https://user-images.githubusercontent.com/16740047/227018309-4d1b9a61-1897-4a0c-9168-ad9f89c621c3.mp4

### CODE REVIEW

**Comment by doralchan** [comment]
@0Calories Is there a gap between the main page filters [Project/Env/Date] and [Category]? Or are they all inline? (Sorry I can't tell at this pixel resolution)

**Comment by 0Calories** [comment]
@doralchan It's easier to see if you fullscreen the video, they're all inline

**Review by udameli — APPROVED** [review]
LGTM

File: static/app/views/issueList/issueCategoryFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  };
+
+  return (
+    <CompactSelect
+      data-test-id="issue-category-filter"
+      options={options}
+      value={selectedOption.value}
+      triggerLabel={
+        <React.Fragment>
+          <Icon /> {renderLabel(selectedOption.value as IssueCategory, true)}
```
[the remark this answers is not in the document]
Comment (0Calories) [reply static/app/views/issueList/issueCategoryFilter.tsx:116]: Ah perfect thanks, I missed this option

**Comment by scefali** [comment]
@0Calories Should we add `profile` now as well that it's been launched at 100%?

File: static/app/views/issueList/issueCategoryFilter.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (0Calories) [reply static/app/views/issueList/issueCategoryFilter.tsx]: That did the trick! Updated

**Comment by 0Calories** [comment]
@scefali Apparently in the backend, profiling and performance issues have the same issue category, so the `Performance` option here will also list profiling issues out of the box

**Comment by scefali** [comment]
> @scefali Apparently in the backend, profiling and performance issues have the same issue category, so the `Performance` option here will also list profiling issues out of the box

@0Calories I believe `profile` is another option as well the API takes but interesting

**Pull request merged**